lib/api/controllers/serverController.js

```diff
@@ -153,42 +161,61 @@ class ServerController extends BaseController {
       .then(() => result);
   }
 
+  /**
+   * Manage errors to be thrown
+   *
+   * @param {...args} ...args
+   * @throws {KuzzleError}
+   */
+
+  throw(...args) {
```

Comment:
this implementation must be exposed by the main `kuzzle` object 

and a much lighter throw method should be put in the BaseController class, which is inherited by all native controllers. An implementation of that method could be:

```
throw(...args) {
  this.kuzzle.throw(this.domain, this.subdomain, ...args);
}
```

(do not forget to add domain and subdomain defaults to the BaseController class constructor)

Reply:
That means I will have to `require` every single json file containing errors definition inside the `kuzzle` object... What do you think?

Reply:
like I said, you'll need a loader (name it `index.js` so you can require the directory directly and the loader will be used)

And in that loader you can either load each file with a static files list, or you can do it dynamically by loading all .js files in the directory. Your call.

Anyway, yes you'll have to load all files at some point. Better than that duplicating the code.

Reply:
I agree with @scottinet with the principle of the `throw` method in the `BaseController` class but we also may want to throw error from core classes

Reply:
no, I said that `throw` was to be exposed by the main kuzzle object
in the BaseController class should only be a throw wrapper, which automatically uses the registered domain/subdomain for that controller (I even provided the code...)

Reply:
So each core class has to implement `throw` as far I understand

Reply:
Why is that?

Reply:
I may want to `throw` an error from the `InternalEngine` for example, so the class has to implement the method and call the `Kuzzle.throw` method  with it's domain and subdomain

Reply:
... or it can use `kuzzle.throw` directly?
A helper is nice in the BaseController class because there are so many controllers and exceptions thrown by them, but otherwise this is not required.